pkg/logs/input/docker/container.go

```diff
@@ -58,6 +60,38 @@ func (c *Container) FindSource(sources []*config.LogSource) *config.LogSource {
 	return bestMatch
 }
 
+// GetSourceShortImageName resolves image name for a container if ContainerCollectAll
+// option is enabled. In that case it returns a new LogSource with the short image name
+// as Source and Service tags. If ContainerCollectAll option is not enable,
+// GetSourceShortImageName returns the source
+func (c *Container) GetSourceShortImageName(source *config.LogSource) *config.LogSource {
```

Comment:
Can we move this method to `Launcher` please ? the `container` package is only meant to be used to match a service and a source, not to create a new source.

Reply:
Can you split the method into two methods, `getShortImageName`, `overrideSource(source, shortImageName)`